pkg/ingester/user_metrics_metadata_purge.go

```diff
@@ -0,0 +1,30 @@
+package ingester
+
+import "time"
+
+const (
+	metadataPurgePeriod = 5 * time.Minute
+)
+
+func (i *Ingester) purgeUserMetricsMetadata() {
+	i.usersMetadataMtx.Lock()
+	usersMetadata := i.usersMetadata
+	i.usersMetadataMtx.Unlock()
```

Comment:
This does nothing.

Unfortunately lock will need to be held during entire `purgeUserMetricsMetadata` method. Iterating the `i.usersMetadata` map while it's possibly being modified by other goroutines is not safe.

Reply:
Good catch, you're completely right - at the end of the day it's just a reference.